### PR TITLE
Consolidate: Pass baseDir & includeDir as options

### PR DESCRIPTION
**Pull request by Kompwu**
Fix issue #1114

### CODE REVIEW

File: src/plugins/ConsolidatePlugin.ts [hunk 1]
```diff
@@ -22,6 +26,8 @@ export class ConsolidatePluginClass implements Plugin {
     this.engine = options.engine;
     this.extension = options.extension || `.${options.engine}`;
     this.useDefault = (options.useDefault !== undefined) ? options.useDefault : true;
+    this.basedir = (options.basedir !== undefined) ? options.basedir : file.context.homeDir;
```
Comment (troynt) [inline src/plugins/ConsolidatePlugin.ts]: file variable does not exist

Comment (Kompwu) [reply src/plugins/ConsolidatePlugin.ts]: @nchanged said that it was...
so idk.

Comment (nchanged) [reply src/plugins/ConsolidatePlugin.ts]: You can't initialize it in the constructor. It has to be done on each file

File: src/plugins/ConsolidatePlugin.ts [hunk 2]
```diff
@@ -53,9 +59,7 @@ export class ConsolidatePluginClass implements Plugin {
     try {
       file.contents = await consolidate[this.engine].render(file.contents, {
         cache: false,
-        filename: 'base',
-        basedir: file.context.homeDir,
-        includeDir: file.context.homeDir
+        filename: 'base'
```
Comment (nchanged) [inline src/plugins/ConsolidatePlugin.ts]: where did the rest of the options go?

Comment (Kompwu) [reply src/plugins/ConsolidatePlugin.ts]: I thought it was just like `useDefault`

File: src/plugins/ConsolidatePlugin.ts [hunk 3]
```diff
@@ -59,7 +57,10 @@ export class ConsolidatePluginClass implements Plugin {
     try {
       file.contents = await consolidate[this.engine].render(file.contents, {
         cache: false,
-        filename: 'base'
+        filename: 'base',; basedir;
```
Comment (troynt) [inline src/plugins/ConsolidatePlugin.ts]: can remove ```; basedir;```

File: src/plugins/ConsolidatePlugin.ts [hunk 4]
```diff
@@ -26,8 +26,6 @@ export class ConsolidatePluginClass implements Plugin {
     this.engine = options.engine;
     this.extension = options.extension || `.${options.engine}`;
     this.useDefault = (options.useDefault !== undefined) ? options.useDefault : true;
-    this.basedir = (options.basedir !== undefined) ? options.basedir : file.context.homeDir;
```
Comment (troynt) [inline src/plugins/ConsolidatePlugin.ts]: this is needed if you plan on referencing it in transform

File: src/plugins/ConsolidatePlugin.ts [hunk 5]
```diff
@@ -26,8 +26,6 @@ export class ConsolidatePluginClass implements Plugin {
...
-    this.includeDir = (options.includeDir !== undefined) ? options.includeDir : file.context.homeDir;
```
Comment (troynt) [inline src/plugins/ConsolidatePlugin.ts]: this is needed if you plan on referencing it in transform

File: src/plugins/ConsolidatePlugin.ts [hunk 6]
```diff
@@ -59,7 +57,10 @@ export class ConsolidatePluginClass implements Plugin {
     try {
       file.contents = await consolidate[this.engine].render(file.contents, {
         cache: false,
-        filename: 'base'
+        filename: 'base',; basedir;
+        basedir: (options.basedir !== undefined) ? options.basedir : file.context.homeDir;
+        includeDir: (options.includeDir !== undefined) ? options.includeDir : file.context.homeDir;
```
Comment (troynt) [inline src/plugins/ConsolidatePlugin.ts]: ```options``` should be ```this```

```basedir: (this.basedir !== undefined) ? this.basedir : file.context.homeDir;```

same for include

Comment (Kompwu) [reply src/plugins/ConsolidatePlugin.ts]: Sorry I'm tired I didn't sleep I code wrong..

File: src/plugins/ConsolidatePlugin.ts [hunk 7]
```diff
@@ -54,8 +58,9 @@ export class ConsolidatePluginClass implements Plugin {
       file.contents = await consolidate[this.engine].render(file.contents, {
         cache: false,
         filename: 'base',
-        basedir: file.context.homeDir,
-        includeDir: file.context.homeDir
+        basedir: this.basedir || file.context.homeDir,
```
Comment (nchanged) [inline src/plugins/ConsolidatePlugin.ts]: where do you assign `basedir`? it's not in the constructor

Comment (pmonty) [reply src/plugins/ConsolidatePlugin.ts]: Looks like its optional so should be fine. I am fussy, could we capitalise the D in basedir as well for consistency 😁

File: src/plugins/ConsolidatePlugin.ts [hunk 8]
```diff
@@ -54,8 +58,9 @@ export class ConsolidatePluginClass implements Plugin {
       file.contents = await consolidate[this.engine].render(file.contents, {
         cache: false,
         filename: 'base',
-        basedir: file.context.homeDir,
-        includeDir: file.context.homeDir
+        basedir: (this.basedir !== undefined) ? this.basedir : file.context.homeDir,
```
Comment (nchanged) [inline src/plugins/ConsolidatePlugin.ts]: `this.baseDir` is never defined

Comment (Kompwu) [reply src/plugins/ConsolidatePlugin.ts]: eh? it is ...

File: src/plugins/ConsolidatePlugin.ts (hunk 7 above)
Comment (Kompwu) [reply src/plugins/ConsolidatePlugin.ts]: Let's capitlise it ! 😄

File: src/plugins/ConsolidatePlugin.ts [hunk 9]
```diff
@@ -26,6 +26,8 @@ export class ConsolidatePluginClass implements Plugin {
     this.engine = options.engine;
     this.extension = options.extension || `.${options.engine}`;
     this.useDefault = (options.useDefault !== undefined) ? options.useDefault : true;
+    this.basedir = options.basedir || file.context.homeDir;
```
[the remark this answers is not in the document]
Comment (Kompwu) [reply src/plugins/ConsolidatePlugin.ts]: I've added tests and everything's fine.
Can you merge it ?

**Comment by nchanged** [comment]
Looking good!

Can you update the [doc](https://github.com/fuse-box/fuse-box/blob/master/docs/plugins/html/ConsolidatePlugin.md) please? After that, I will merge it

**Comment by Kompwu** [comment]
@nchanged Docs added :)

**Pull request merged**